app/src/org/commcare/dalvik/activities/FormRecordListActivity.java

```diff
@@ -502,105 +497,15 @@ public boolean onPrepareOptionsMenu(Menu menu) {
-                                task.execute();
-                                break;
-                            case DataPullTask.UNKNOWN_FAILURE:
-                                Toast.makeText(receiver, "Failure retrieving or processing data, please try again later...", Toast.LENGTH_LONG).show();
-                                break;
-                            case DataPullTask.AUTH_FAILED:
-                                Toast.makeText(receiver, "Authentication failure. Please logout and resync with the server and try again.", Toast.LENGTH_LONG).show();
-                                break;
-                            case DataPullTask.BAD_DATA:
-                                Toast.makeText(receiver, "Bad data from server. Please talk with your supervisor.", Toast.LENGTH_LONG).show();
-                                break;
-                            case DataPullTask.CONNECTION_TIMEOUT:
-                                Toast.makeText(receiver, "The server took too long to generate a response. Please try again later, and ask your supervisor if the problem persists.", Toast.LENGTH_LONG).show();
-                                break;
-                            case DataPullTask.SERVER_ERROR:
-                                Toast.makeText(receiver, "The server had an error processing your data. Please try again later, and contact technical support if the problem persists.", Toast.LENGTH_LONG).show();
-                                break;
-                            case DataPullTask.UNREACHABLE_HOST:
-                                Toast.makeText(receiver, "Couldn't contact server, please check your network connection and try again.", Toast.LENGTH_LONG).show();
-                                break;
-                        }
-                    }
-
-                    @Override
-                    protected void deliverUpdate(FormRecordListActivity receiver, Integer... update) {
-                        switch (update[0]) {
-                            case DataPullTask.PROGRESS_AUTHED:
-                                receiver.updateProgress("Authed with server, downloading forms" +
-                                                (update[1] == 0 ? "" : " (" + update[1] + ")"),
-                                        DataPullTask.DATA_PULL_TASK_ID);
-                                break;
-                        }
-                    }
-
-                    @Override
-                    protected void deliverError(FormRecordListActivity receiver, Exception e) {
-                        receiver.taskError(e);
-                    }
-                };
-                pull.connect(this);
-                pull.execute();
+                ArchivedFormRemoteRestore<FormRecordListActivity> archiveRemoteForms =
+                        new ArchivedFormRemoteRestore<>(this, this, platform);
+                archiveRemoteForms.pullArchivedFormsFromServer(source);
```

Comment:
Launches task that is only used by developers to saved load forms manually onto phones.